### ISSUE TITLE
[bench request]: Add JevK5 v0.1 (Qwen3.5-4B + distilled LoRA, native option-letter readout, in-process adapter)

### ISSUE THREAD

**Issue by allebee** [fstandhartinger/jevbench#31]
Hi, I would like to submit **JevK5** to JevBench.

JevK5 is an open decision model: Qwen3.5-4B with a LoRA distilled from a thinking teacher, read in
one forward pass with SemIf's protocol (the restricted softmax over the declared options'
answer-letter logits at the last position), divided by one calibration temperature. `native`
probabilities, zero generated tokens. It runs as an in-process adapter, so the held-out items stay
on your machine, and it records one CUDA graph per padded input length, so a decision takes
~13 ms on an H100.

## System Details
- **Name:** JevK5 v0.1
- **Repo:** https://github.com/allebee/jevk5 — tag `v0.1.0` (commit `523aabe`)
- **Weights:** `alibiserikbay/JevK5`, bf16, hub revision `c5be5f3131335490c28578829a041b07c2466c80` (Qwen3.5-4B + LoRA r16 on
  the attention projections, merged)
- **License:** Apache-2.0 (code and weights); base Qwen3.5-4B Apache-2.0; prompt and readout
  adapted from SemIf (MIT, attributed in `NOTICE`)
- **Readout:** identical option mapping to `semif_direct`; the distribution is divided by
  T = 1.367 (`jevk5_config.json`)
- **Training data:** 1,635 decisions written and double-checked by Qwen3.6-27B with thinking
  (Apache-2.0) plus 1,635 human-labelled items from public datasets. **No JevBench item, public or
  held out, and no output of Jev was used for training, tuning or selection.** The temperature
  was fitted on teacher questions from three held-out domains, not on JevBench.
- **Output tokens:** 0 per decision

## How to Reproduce / Run
In-process adapter, no server. `bench/jevk5_direct.py` drops into `jevbench/adapters/`;
`bench/jevbench-registration.patch` applies the registration against `0caa1d0`. Needs one CUDA
GPU with ~10 GB free (≈ 8.5 GB in bf16).

```bash
git clone https://github.com/allebee/jevk5 && cd jevk5 && git checkout v0.1.0
pip install -e ".[fast]"
cp bench/jevk5_direct.py <jevbench>/jevbench/adapters/
( cd <jevbench> && git apply <jevk5>/bench/jevbench-registration.patch )
cd <jevbench> && JEVBENCH_WARM_LOAD=1 python -m jevbench.cli run --adapter jevk5_direct \
    --endpoint alibiserikbay/JevK5 --revision c5be5f3131335490c28578829a041b07c2466c80 --model jevk5-v0.1 ...
```
Full steps and the system card are in `bench/SUBMISSION.md`. Inputs over 4,096 tokens and
questions with more than 16 options are refused, never truncated, so the runner records them
and continues.

## Reference Local Run (Public Items)
231 public items through the official runner at `0caa1d0` (`results/public231/`): 231/231
strict-valid, 0 failures.

| Split | n | Accuracy | ECE | p50 raw | p95 raw |
|---|---:|---:|---:|---:|---:|
| easy | 48 | **1.000** | 0.026 | 13.5 ms | 14.9 ms |
| original (standard) | 72 | 0.958 | 0.073 | 13.6 ms | 14.8 ms |
| hard (public half) | 111 | **0.676** | **0.082** | 30.1 ms | 160.9 ms |

For comparison, the same base model and prompt without the LoRA and the temperature (its public
outcomes match SemIf's official ones on 231/231) scores 1.000 / 0.986 / 0.613, hard-tier ECE 0.117.

- **Latency:** one H100, in-process, batch 1, CUDA graphs; your ×2 + 0.15 s self-hosted
  adjustment applies
- **Cost basis:** same weights class as SemIf, deepinfra Qwen3.5-4B $0.03/M in, with measured
  input tokens (164 easy / 168 standard / 1,274 hard per decision) and 0 output tokens
- **Known weak spots:** two standard-tier items the untrained model gets right are wrong after
  training; probability fidelity mean TVD 0.296 on the 10 public exact-distribution items

A v0.2 trained on more teacher data is planned within a few days; I will post it as a comment
here rather than open a new issue. Thanks for running the benchmark!

**Comment by fstandhartinger** [comment]
Thanks — this is logged for reachability checks and the JevBench measurement queue. We’ll only make benchmark claims after our own run.

- Harold

**Comment by allebee** [comment]
**Before your run: please use tag `v0.1.1` (commit `06bfa58`) instead of `v0.1.0`.**

One change, in the adapter only. `jevk5_direct` now answers inputs up to 16,384 tokens; v0.1.0 refused anything over 4,096. The hard tier's policy documents run to about 6k tokens, and under our prompt the longest public one is already 4,033 tokens. Held-out items over the old limit would have counted as wrong because of a limit we set, not one the model has. Inputs past the largest CUDA graph (4,096 tokens) run the same model eagerly. We checked inputs of 8k, 12.7k and 27.8k tokens: same answer.

Unchanged: the weights and hub revision `c5be5f3131335490c28578829a041b07c2466c80`, the prompt, the temperature (1.367), and every answer at or under 4,096 tokens, so the public-item table above stands. The registration patch applies unchanged to `v1.3.0` and current `main`.

```bash
git clone https://github.com/allebee/jevk5 && cd jevk5 && git checkout v0.1.1
```

All other steps are as in the request above.

**Comment by allebee** [comment]
**v0.2 is up, and it supersedes v0.1.1.** The issue title and body now describe it; the v0.1 details are kept at the bottom of the body. If v0.1.1 is already in your queue, either is fine to run, but v0.2 is the one we stand behind.

What changed: the same recipe on twice the teacher data (3,272 questions instead of 1,635, same eleven families plus a 9-question pilot of a probability family whose answers are exact distributions), and the temperature refitted the same way, 1.367 → 1.532.

| Split | n | v0.1 | **v0.2** | untrained |
|---|---:|---:|---:|---:|
| easy | 48 | 1.000 | **1.000** | 1.000 |
| original (standard) | 72 | 0.958 | 0.958 | 0.986 |
| hard (public half) | 111 | 0.676 | **0.739** | 0.613 |
| hard-tier ECE | | 0.082 | **0.066** | 0.117 |
| mean TVD, 10 probability items | | 0.296 | **0.196** | 0.298 |

231/231 strict-valid, 0 failures, through your runner with the packaged adapter. On the hard tier v0.2 fixes 21 of the untrained model's items and breaks 7 (McNemar p = 0.013); against v0.1, 9 fixed and 2 broken. Per-item results are in `results/public231/jevk5-v0.2.jsonl`.

```bash
git clone https://github.com/allebee/jevk5 && cd jevk5 && git checkout v0.2.0
# weights: alibiserikbay/JevK5 @ 3c673298eb7f2dc7cb98019262c55b1fac01a0bc
```

Two changes many entrants here use were measured and **rejected**, both on held-out data only (`training/temp_choice.py`, `training/order_avg.py`): a temperature per question type (better on the teacher's held-out domains, ECE 0.032 vs 0.034, clearly worse on our hand-written hard set, 0.129 vs 0.091) and averaging two option orders (lower ECE but lower accuracy, 0.799 → 0.794, at twice the work). Disclosure unchanged: no JevBench item, public or held out, and no output of Jev was used for training, tuning or selection; the public items are only reported.

**Comment by allebee** [comment]
Thank you for the v1.4 run, and for the overlap scan — your note about the shared instruction is correct, and it made us scan our own published files properly. We found a second echo you did not flag, so both are below.

**What the scan found in `training/hard_dev_items.py`** (our 65-item hand-written calibration set):

1. The generic judge instruction you flagged, `"Does the response fully and correctly satisfy the request?"`, identical to the instruction in eight public hard items.
2. One date item whose rule wording followed a public item's closely: "cover starts on the delivery date and ends 18 months later / if the end month has no day with the same number".

Both are rewritten as of commit `dd002db`. An 8-word-sequence scan of every published file against all 231 public items now returns nothing.

**What it affected, stated plainly.** That set has never been training data; the teacher wrote every training question from scratch. It was, however, one of the held-out checks behind a published choice: one shared temperature (1.532) against one per question type. It argued for the shared temperature we shipped — and the per-type option measures **better** on the public items (hard-tier ECE 0.046 against 0.066). So the echo pushed us toward the weaker configuration on your items rather than the stronger one. A temperature never reorders a distribution, so no accuracy number in our tables or in your run changes.

Our earlier sentence "no JevBench item, public or held out, was used for training, tuning or selection" was therefore too clean for that calibration set, and the repository now carries the correction in `CHANGELOG.md` rather than a quiet edit.

On the run itself: the sealed set is a good addition, and our 33.1% on it against 85.3% public is the honest measure of what a single forward pass with no generated tokens cannot do. The family breakdown is useful — paraphrase robustness 0.143 against Jev's 0.643 is the clearest thing for us to fix, and it is what v0.3 will target.

**Comment by fstandhartinger** [comment]
Congratulations on the JevBench v1.4 release, allebee! Thanks for building and sharing your system.

- JevK5 v0.2.0: #2 at 62.04

JevK5 v0.2.0 is newly public in v1.4. V1.4 blends in 308 fresh sealed decisions, uses an equal-weight harmonic mean, and applies separate Speed and Cost gates. For your leading row, sealed accuracy was 33.1% versus 85.3% on the public slice. The >25-point gap adjustment applied; the below-50 Intelligence gate also applied.

Full board and method: https://benchmarkheaven.com/jev-models
